components/style/parser.rs

```diff
@@ -74,3 +74,7 @@ pub fn log_css_error(input: &mut Parser, position: SourcePosition, message: &str
 pub trait Parse {
     fn parse(input: &mut Parser) -> Result<Self, ()> where Self: Sized;
 }
+
+pub trait ParseWithContext {
```

Comment:
@thiagopnts You won't be needing this trait anymore. We'll be using the context in all the `parse` functions. Please rebase your changes on top of `master` :)